htlcswitch/switch.go

```diff
@@ -754,66 +755,90 @@ func (s *Switch) handleLocalDispatch(pkt *htlcPacket) error {
 	// User have created the htlc update therefore we should find the
 	// appropriate channel link and send the payment over this link.
 	if htlc, ok := pkt.htlc.(*lnwire.UpdateAddHTLC); ok {
-		// Try to find links by node destination.
-		s.indexMtx.RLock()
-		link, err := s.getLinkByShortID(pkt.outgoingChanID)
-		s.indexMtx.RUnlock()
+		// Try to get the link that the HTLC should be forwarded outward on,
+		// and check that it satisfies the link's policy.
+		link, err := s.handleLocalAddHTLC(pkt, htlc)
```

Comment:
Isolating code move in its own commit makes review easier

Reply:
Will do, I'm going to split in to multiple PRs as suggested somewhere else here and split up the notify commits as well (into sends/receives and forwards where possible)